### PR TITLE
Created Sonrai Security Solution for Azure Sentinel

### PR DESCRIPTION
**Pull request by Cameron-Dahr-sonrai**
## Proposed Changes

  - Added Sonrai Security Solution to Azure Sentinel

### CODE REVIEW

File: Solutions/SonraiSecurity/Workbooks/Images/Logo/Sonrai.svg [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (NikTripathi) [inline Solutions/SonraiSecurity/Workbooks/Images/Logo/Sonrai.svg:1]: Size of the logo should be less than 5Kb.Please refer the [guidelines](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/ReadMe.md)

File: Sample Data/Custom/Sonrai_CL.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
+[
```
Comment (NikTripathi) [inline Sample Data/Custom/Sonrai_CL.json:1]: File name for sample data should match with custom table name. So in our case it should be **Sonrai_Tickets_CL** .

File: Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
+{
+  "id": "SonraiDataConnector",
+  "title": "Sonrai Data Connector",
+  "publisher": "Sonrai",
+  "descriptionMarkdown": "Use this data connector to integrate with Sonrai Security and get Sonrai tickets sent directly to Azure Sentinel.",
+  "graphQueries": [
+    {
+      "metricName": "Sonrai Tickets",
+      "legend": "Sonrai_Ticket_Logs",
```
Comment (NikTripathi) [inline Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json:9]: Legend should be same as data type name.

File: Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      "baseQuery": "Sonrai_Ticket_Logs"
```
Comment (NikTripathi) [inline Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json:10]: Can you please tell us what is Sonrai_Ticket_Logs? If it is a custom table please provide sample data.

**Review by NikTripathi — CHANGES_REQUESTED** [review]

File: Logos/Sonrai.svg [hunk 5]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (Cameron-Dahr-sonrai) [reply Logos/Sonrai.svg:1]: We have managed to compress our logo down to 6Kb, would this be acceptable?

[the remark this answers is not in the document]
Comment (NikTripathi) [reply Logos/Sonrai.svg:1]: No, The limit is 5 kb. Above this the validation will not succeed.

[the remark this answers is not in the document]
Comment (Cameron-Dahr-sonrai) [reply Logos/Sonrai.svg]: We have replaced the logo with an alternate under 5kb

File: Sample Data/Custom/Sonrai_CL.json (hunk 2 above)
Comment (Cameron-Dahr-sonrai) [reply Sample Data/Custom/Sonrai_CL.json]: Renamed the file to Sonrai_Tickets_CL.json

File: Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json (hunk 4 above)
Comment (Cameron-Dahr-sonrai) [reply Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json]: Changed Sonrai_Ticket_Logs to Sonrai_Tickets_CL

**Comment by NikTripathi** [comment]
@ashwin-patil can you please review the analytical rules.

File: Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    {
+      "metricName": "Sonrai Tickets",
+      "legend": "Sonrai_Tickets_CL",
+      "baseQuery": "Sonrai_Tickets_CL"
+    }
+  ],
+  "sampleQueries": [
+    {
+      "description" : "Query for tickets with PermissionList violations",
+      "query": "Sonrai_Tickets_CL \n| where evidence_policyEvidence_label_s == \"PermissionList\"\n| limit 10"
```
Comment (v-maudan) [inline Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json:16]: @Cameron-Dahr-sonrai, Given sample query is failing at column 'evidence_policyEvidence_label_s' , Can you please check and update? Thanks!
image

**Review by v-maudan — CHANGES_REQUESTED** [review]

File: Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json (hunk 6 above)
Comment (Cameron-Dahr-sonrai) [reply Solutions/SonraiSecurity/Data Connectors/Connector_REST_API_Sonrai.json]: Modified this query

**Comment by v-rucdu** [comment]
@ashwin-patil : Could you please review Analytical Rules?
Thanks!!!!

**Comment by Cameron-Dahr-sonrai** [comment]
@ashwin-patil Just let me know if there are any changes you'd like for the analytical rules

**Review by ashwin-patil — CHANGES_REQUESTED** [review]
@Cameron-Dahr-sonrai  Sorry for the delay. Analytics queries looks good and straight-forward atomic. 

Couple of minor changes -
- Can you update sample data to match with the schema in the query e.g. `action_d` instead of `action` ? 
- Also if possible, upload csv version of sample data rather than json array to easily test the rules.  

once these changes checked in , good to merge.

**Comment by Cameron-Dahr-sonrai** [comment]
@ashwin-patil I left the old Sonrai_Tickets_CL.json data as it was since it already contains the unflattened format of the data as we send it to Sentinel, but I've created a new sample data file Sonrai_Tickets_CL.csv which contains a csv version of the sample data with the schema requested to match the queries (e.g action_d). If you'd also like me to change the json version, just let me know.

**Review by ashwin-patil — APPROVED** [review]

**Review by NikTripathi — APPROVED** [review]

**Comment by Cameron-Dahr-sonrai** [comment]
@v-maudan If the PR looks good, could you approve the changes?

**Pull request merged**